cmd/collector/app/server/zipkin.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/cors"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/collector/app/handler"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/zipkin"
+	"github.com/jaegertracing/jaeger/pkg/healthcheck"
+)
+
+// ZipkinServerParams to construct a new Jaeger Collector Zipkin Server
+type ZipkinServerParams struct {
+	Port            int
+	Handler         handler.ZipkinSpansHandler
+	RecoveryHandler func(http.Handler) http.Handler
+	AllowedOrigins  string
+	AllowedHeaders  string
+	HealthCheck     *healthcheck.HealthCheck
+	Logger          *zap.Logger
+}
+
+// StartZipkinServer based on the given parameters
+func StartZipkinServer(params *ZipkinServerParams) (*http.Server, error) {
+	if params.Port == 0 {
+		return nil, nil
+	}
+
+	httpPortStr := ":" + strconv.Itoa(params.Port)
+	params.Logger.Info("Listening for Zipkin HTTP traffic", zap.Int("zipkin.http-port", params.Port))
+
+	listener, err := net.Listen("tcp", httpPortStr)
+	if err != nil {
+		return nil, err
+	}
+
+	server := &http.Server{Addr: httpPortStr}
+	if err := serveZipkin(server, listener, params); err != nil {
+		return nil, err
+	}
+
+	return server, nil
+}
+
+func serveZipkin(server *http.Server, listener net.Listener, params *ZipkinServerParams) error {
```

Comment:
error is not being used

Reply:
All other `serve` functions are returning an error, even though only the Thrift one actually returns an error. I would keep it, so that callers can handle it once we decide to unify the signatures under an interface.

Reply:
The API should be clear and return what is needed for a caller to know and handle.

Reply:
Alright, removed it. If we refactor it and use a common interface, we can add it back.